File: backend/src/main/kotlin/com/study/restaurantapp/model/Order.kt
package com.study.restaurantapp.model

import org.springframework.data.annotation.Id
import org.springframework.data.mongodb.core.mapping.Document

@Document
data class Order(
        @Id
        val id: String,
        val client: User,
        val items: List<MenuPosition>,
        val status: OrderStatus
)

File: frontend/restaurant-app/src/app/models/order.ts
import { User } from './user';
import {MenuPosition} from "./menu-positions";


export enum OrderStatus {
  IN_PROGRESS = "IN_PROGRESS",
  READY_FOR_PICKUP = "READY_FOR_PICKUP"
}

export interface Order {
  id: string;
  client: User;
  items: MenuPosition[];
  status: OrderStatus;
}
